rfd/0144-client-tools-updates.md

```diff
@@ -0,0 +1,296 @@
+automatically updated. https://goteleport.com/docs/upgrading/automatic-updates
+```
+
+If the version of client tools is 2 (or more) versions lower than the version
+advertised by the cluster or 1 (or more) version greater than the version
+advertised by the cluster, an error will be shown and will require the user to
+use the `--skip-version-check` flag.
+
+```
+$ tsh login --proxy=proxy.example.com
+ERROR: Client tools are N major versions out of date, update to vX.Y.Z.
+
+Your cluster requires {tsh,tctl} vX.Y.Z. Update Teleport from
+https://goteleport.com/download or your system package manager.
+
+Enroll in automatic updates to keep client tools like tsh and tctl
+automatically updated. https://goteleport.com/docs/upgrading/automatic-updates
+
+Use the "--skip-version-check" flag to bypass this check and attempt to connect
+to this cluster.
+```
+
+#### Self-managed client tools updates
+
+Cluster administrators that want to self-manage client tools updates will be
+able to get and watch for changes to client tools versions which can then be
+used to trigger other integrations (using MDM software like Jamf) to update the
+installed version of client tools on endpoints.
+
+```
+$ tctl autoupdate watch
+{"tools_version": "1.0.0"}
+{"tools_version": "1.0.1"}
+{"tools_version": "2.0.0"}
+
+[...]
+```
+
+```
+$ tctl autoupdate get
+{"tools_version": "2.0.0"}
+```
+
+##### Cluster configuration
+
+Enrollment of clients in automatic updates will be enforced at the cluster
+level.
+
+The `cluster_maintenance_config` resource will be updated to allow cluster
+administrators to turn client tools automatic updates `on` or `off`.
+A `autoupdate_version` resource will be added to allow cluster administrators
+to manage the version of tools pushed to clients.
+
+> [!NOTE]
+> Client tools configuration is broken into two resources to [prevent
+> updates](https://github.com/gravitational/teleport/blob/master/lib/modules/modules.go#L332-L355)
+> to `autoupdate_version` on Cloud.
+>
+> While Cloud customers will be able to use `cluster_maintenance_config` to
+> turn client tools automatic updates `off` and self-manage updates, they will
+> not be able to control the version of client tools in `autoupdate_version`.
+> That will continue to be managed by the Teleport Cloud team.
+
+Both resources can either be updated directly or by using `tctl` helper
+functions.
+
+```yaml
+kind: cluster_maintenance_config
+spec:
+  # tools_auto_update allows turning client tools updates on or off at the
+  # cluster level. Only turn client tools automatic updates off if self-managed
+  # updates are in place.
+  tools_auto_update: on|off
+
+  [...]
+```
+```
+$ tctl autoupdate update --set-tools-auto-update=off
+Automatic updates configuration has been updated.
+```
+
+By default, all Cloud clusters will be opted into `tools_auto_update: on`. All
+self-hosted clusters will be opted into `tools_auto_update: off`.
```

Comment:
I'll add this how to use client tools updates to our documentation, but am thinking about waiting a bit before pushing it to self-hosted customers.

The main reason I am hesitating is because self-hosted customers often have multiple clusters they connect to, and automatic updates in their current form won't work well for them.

We want to address some form of simple or automatic updates for self-hosted customers as well, but we'll do it as Phase 3.